jabgui/src/main/java/org/jabref/gui/preferences/JabRefGuiPreferences.java

```diff
@@ -998,15 +1000,32 @@ private NameDisplayPreferences.DisplayStyle getNameDisplayStyle() {
 
     // region: Main table, main table column, and search dialog column preferences
 
+    private MainTablePreferences getMainTablePreferencesFromBackingStore(MainTablePreferences defaults) {
+        return new MainTablePreferences(
+                getMainTableColumnPreferences(),
+                getBoolean(AUTO_RESIZE_MODE, defaults.getResizeColumnsToFit()),
+                getBoolean(EXTRA_FILE_COLUMNS, defaults.getExtraFileColumnsEnabled())
+        );
+    }
+
+    private ColumnPreferences getMainTableColumnPreferencesFromBackingStore(ColumnPreferences defaults) {
+        List<String> storedColumnNames = getStringList(COLUMN_NAMES);
+
+        if (storedColumnNames.isEmpty()) {
+            return defaults;
+        }
+
+        List<MainTableColumnModel> columns = getColumns(COLUMN_NAMES, COLUMN_WIDTHS, COLUMN_SORT_TYPES, ColumnPreferences.DEFAULT_COLUMN_WIDTH);
+        List<MainTableColumnModel> columnSortOrder = getColumnSortOrder(COLUMN_SORT_ORDER, columns);
+        return new ColumnPreferences(columns, columnSortOrder);
+    }
+
```

Comment:
Please move these methods below the getXXXPreferences methods to keep consistency in the file with other preferences. Helps reading and understanding the code.